src/backend/marsha/core/serializers/base.py

```diff
@@ -37,26 +37,41 @@
 KEY_REGEX = re.compile(KEY_PATTERN)
 
 
-def get_video_cloudfront_url_params(video_id):
+def get_resource_cloudfront_url_params(resource_kind: str, resource_id: str):
     """
-    Generate the policy and sign it to allow to access to all resources for a given video id.
+    Generate the policy and sign it to allow to access to all sub resources
+    for a given resource id.
+    Parameters
+    ----------
+    resource_kind: str
+        Only used to define the cache key.
+    resource_id: str
+        The "parent" resource ID
     """
     resource = (
-        f"{settings.AWS_S3_URL_PROTOCOL}://{settings.CLOUDFRONT_DOMAIN}/{video_id}/*"
+        f"{settings.AWS_S3_URL_PROTOCOL}://{settings.CLOUDFRONT_DOMAIN}/{resource_id}/*"
     )
-    cache_key = f"cloudfront_signed_url:video:{video_id}:{resource}"
+    cache_key = f"cloudfront_signed_url:{resource_kind}:{resource_id}:{resource}"
     date_less_than = timezone.now() + timedelta(
         seconds=settings.CLOUDFRONT_SIGNED_URLS_VALIDITY
     )
     if (params := cache.get(cache_key)) is None:
         params = cloudfront_utils.generate_cloudfront_urls_signed_parameters(
-            resource, date_less_than=date_less_than
+            resource,
+            date_less_than=date_less_than,
         )
         cache.set(cache_key, params, settings.CLOUDFRONT_SIGNED_URL_CACHE_DURATION)
 
     return params
 
 
+def get_video_cloudfront_url_params(video_id):
+    """
+    Generate the policy and sign it to allow to access to all resources for a given video id.
+    """
+    return get_resource_cloudfront_url_params("video", video_id)
```

Comment:
Maybe use `Video.S3_IDENTIFIER` here ?
```suggestion
    return get_resource_cloudfront_url_params(Video.S3_IDENTIFIER, video_id)
```
And move this function in `src/backend/marsha/core/serializers/video.py` ?